### PR TITLE
6327 soliditySha3 with BigInt

### PR DESCRIPTION
**Pull request by jdevcs**
## Description

https://github.com/web3/web3.js/issues/6327

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

**Review by avkos — APPROVED** [review]

File: packages/web3-utils/src/hash.ts [hunk 1]
```diff
@@ -159,8 +159,12 @@ const getType = (arg: Sha3Input): [string, EncodingTypes] => {
 		type = 't' in arg ? arg.t : arg.type;
 		value = 'v' in arg ? arg.v : arg.value;
 
-		// otherwise try to guess the type
-	} else {
+		type = type.toLowerCase() === 'bigint' ? 'int' : type;
```
Comment (luu-alex) [inline packages/web3-utils/src/hash.ts:162]: Looks like from muhammads' [example  ](https://github.com/web3/web3.js/issues/6327) number strings also need to be converted to int

File: packages/web3-utils/test/fixtures/hash.ts [hunk 2]
```diff
@@ -304,3 +304,48 @@ export const elementaryNameValidData: [any, string][] = [
...
+
+	[['0x70696e67', 0], '0xe54a278c69f07b6b4f0736dc55c389cd2d3f31365b090c9f76a414fb51552c53'],
+	[
+		['0x70696e67', BigInt(0)],
+		'0xe54a278c69f07b6b4f0736dc55c389cd2d3f31365b090c9f76a414fb51552c53',
+	],
+
+	[['0x70696e67', 10], '0x418e921c5c859d5f560b89ad03a6672907d66de336392a5178ea69281feff40a'],
+	[
+		['0x70696e67', BigInt(10)],
```
Comment (luu-alex) [inline packages/web3-utils/test/fixtures/hash.ts:341]: add string test case using `"10"`

**Comment by luu-alex** [comment]
LGTM other than my comments

**Review by luu-alex — APPROVED** [review]

File: packages/web3-utils/src/hash.ts (hunk 1 above)
Comment (jdevcs) [reply packages/web3-utils/src/hash.ts:162]: I [think](https://github.com/web3/web3.js/issues/6327#issuecomment-1667521008) these are diff data types and 4.x is aligned with solidity , as in solidity sha3 output is different for string vs number.
```
contract ShaHashTest {
	    function func10() external pure returns (bytes32) {
        		return keccak256(abi.encodePacked(int(10))) ;}

        function funcStr10() external pure returns (bytes32) {
        		return keccak256(abi.encodePacked(string("10"))) ;}
}
```
for `int` 10 its : 0xc65a7bb8d6351c1cf70c95a316cc6a92839c986682d98bc35f958f4883f9d2a8
and for `bytes/string` '10' its : 0x1a192fabce13988b84994d4296e6cdc418d55e2f1d7f942188d4040b94fc57ac

**Pull request merged**